core/services/chainlink/config_dump.go

```diff
@@ -0,0 +1,270 @@
+package chainlink
+
+import (
+	"encoding/csv"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/pelletier/go-toml/v2"
+	ocrnetworking "github.com/smartcontractkit/libocr/networking"
+
+	"github.com/smartcontractkit/chainlink/core/config"
+	"github.com/smartcontractkit/chainlink/core/config/envvar"
+	"github.com/smartcontractkit/chainlink/core/config/parse"
+	tcfg "github.com/smartcontractkit/chainlink/core/config/toml"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+func (app *ChainlinkApplication) ConfigDump() (string, error) {
+	var c Config
+
+	if err := c.loadChainsAndNodes(app.Chains); err != nil {
+		return "", err
+	}
+
+	c.loadLegacyEVMEnv()
+
+	c.loadLegacyCoreEnv()
+
+	b, err := toml.Marshal(c)
+	if err != nil {
+		return "", err
+	}
+	//TODO what about secrets?
+
+	return string(b), nil
+}
+
+//TODO doc
+func (c *Config) loadChainsAndNodes(chains Chains) error {
+	//TODO copy chains and nodes from database
+	//TODO must iterate over pure ORM methods for all chains and nodes
+	//TODO even disabled chains?
```

Comment:
What about in the future, if a node operator wants to disable a chain temporarily? Will we require them to comment it out? I am concerned this could be error prone, since if you miss some things at the bottom they could just implicitly be applied to the previously listed chain.
Consider a node op running on main and test, that biffs disabling test:
```toml
[[EVM]]
ChainID = 1

...

# [[EVM]]
# ChainID = 42
# ....
# ....
# ....

[EVM.BlockHistoryEstimator]
TransactionPercentile = 100

[[EVM.Nodes]]
...
```
Now the testnet's `BlockHistoryEstimator` is unwittingly applied to mainnet, and the client node pool includes testnet nodes (fortunately we validate chain id, but will still produce confusing log noise).

Alternatively, an explicit field could make this clear, simple and foolproof:
```toml
[[EVM]]
Enabled = false
ChainID = 1
```

Reply:
Wired up with `Enabled = false` for now, since that is trivial to do compared to producing toml comments, but can revisit if that is important. My gut is that we should be preserving the data and automating as much as possible, but on the other hand this data may just be noise if they never plan to re-enable :shrug:

Reply:
We could require an explicit `Enabled = true` for consistency with the other tables, but on the other hand, this seems like a slightly difference case and it doesn't have the 'empty table' problem, because they must include `ChainID` even if they use all other defaults:
```toml
[[EVM]]
ChainID = 1

[[EVM]]
ChainID = 42
Enabled = false
```